### ISSUE TITLE
TestPathCommand and several others removed in PowerShell 7.2 SDK

### ISSUE THREAD

**Issue by colinblaise** [PowerShell/PowerShell#16862]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Using C# SDK, I updated `Microsoft.PowerShell.Commands.Management` from `7.1.5` to `7.2.1`.

The SDK no longer has references to the following commands:

1. JoinPathCommand
2. TestPathCommand
3. GetChildItemCommand
4. GetItemCommand
5. SetItemCommand

If this was intentional, how should I be referencing these commands?

### Expected behavior

```console
I need to import the commands listed under steps to reproduce
```

### Actual behavior

```console
I can now longer use the commands listed under steps to reproduce
```

### Environment data

```powershell
Key                       Value
---                       -----
PSVersion                 7.2.1
PSEdition                 Core
GitCommitId               7.2.1
OS                        Linux 5.10.60.1-microsoft-standard-WSL2 #1 SMP Wed Aug 25 23:20:18 UTC 2021
Platform                  Unix
PSCompatibleVersions      {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion 2.3
SerializationVersion      1.1.0.1
WSManStackVersion         3.0
```

**Comment by colinblaise** [comment]
Related - https://github.com/PowerShell/PowerShell/issues/16141

**Comment by dkattan** [comment]
Further, if you try to include Microsoft.PowerShell.Commands.Management from NuGet you get the error:

Assembly 'Microsoft.PowerShell.Commands.Management' with identity 'Microsoft.PowerShell.Commands.Management, Version=7.3.0.2, Culture=neutral, PublicKeyToken=31bf3856ad364e35' uses 'System.Management.Automation, Version=7.3.0.2, Culture=neutral, PublicKeyToken=31bf3856ad364e35' which has a higher version than referenced assembly 'System.Management.Automation' with identity 'System.Management.Automation, Version=7.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'

**Comment by dkattan** [comment]
@SeeminglyScience 

There are 2 problems:
1. Microsoft.PowerShell.SDK is missing an assembly reference to Microsoft.PowerShell.Commands.Management.dll, Microsoft.PowerShell.Commands.Diagnostics, Microsoft.PowerShell.Security, and Microsoft.WSMan.Management
2. Microsoft.PowerShell.Commands.Management references System.Management.Automation.dll 7.3.0.0 while Microsoft.PowerShell.SDK references 7.3.0.500

Personally, I don't really care about 2 because I only needed it due to the missing reference in Microsoft.PowerShell.SDK

## Problem 1

![image](https://user-images.githubusercontent.com/1424395/205742617-9983c406-67a0-4fc3-81a3-0470e1cbe715.png)

![image](https://user-images.githubusercontent.com/1424395/205742640-c32fbaae-3955-41b6-966e-d8531847d50e.png)


I believe to fix it we need to the missing assemblies to the $supportedRefList array to this block of code:
https://github.com/PowerShell/PowerShell/blob/07175ae0ff8eb7306fe0b0fc7d19bdef4fbf2d67/tools/packaging/packaging.psm1#L2265-L2267

and also to
https://github.com/PowerShell/PowerShell/blob/07175ae0ff8eb7306fe0b0fc7d19bdef4fbf2d67/tools/packaging/packaging.psm1#L2456-L2460

## Problem 2
If you are interested in also fixing the mismatch in problem 2, if you add a hintpath you can get Visual Studio to tell you more information about the issue:

Before
```xml
<ItemGroup>
    <PackageReference Include="Microsoft.PowerShell.SDK" Version="7.1.5" />
    <PackageReference Include="Microsoft.PowerShell.Commands.Management" Version="7.1.5"/>
</ItemGroup>
```

After
```xml
<ItemGroup>
    <PackageReference Include="Microsoft.PowerShell.SDK" Version="7.1.5" />
    <PackageReference Include="Microsoft.PowerShell.Commands.Management" Version="7.1.5">
      <GeneratePathProperty>True</GeneratePathProperty>
    </PackageReference>
  </ItemGroup>

  <ItemGroup>
    <Reference Include="Microsoft.PowerShell.Commands.Management"> 
<HintPath>$(PkgMicrosoft_powershell_commands_management)\runtimes\unix\lib\net6.0\Microsoft.PowerShell.Commands.Management.dll</HintPath>
    </Reference>
  </ItemGroup>
```


```
Found conflicts between different versions of "System.Management.Automation" that could not be resolved.
There was a conflict between "System.Management.Automation, Version=7.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" and "System.Management.Automation, Version=7.3.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35".
    "System.Management.Automation, Version=7.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" was chosen because it was primary and "System.Management.Automation, Version=7.3.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35" was not.
    References which depend on "System.Management.Automation, Version=7.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" [C:\Users\DKattan.IN\.nuget\packages\system.management.automation\7.3.0\ref\net7.0\System.Management.Automation.dll].
        C:\Users\DKattan.IN\.nuget\packages\system.management.automation\7.3.0\ref\net7.0\System.Management.Automation.dll
          Project file item includes which caused reference "C:\Users\DKattan.IN\.nuget\packages\system.management.automation\7.3.0\ref\net7.0\System.Management.Automation.dll".
            C:\Users\DKattan.IN\.nuget\packages\system.management.automation\7.3.0\ref\net7.0\System.Management.Automation.dll
    References which depend on "System.Management.Automation, Version=7.3.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35" [].
        C:\Users\DKattan.IN\.nuget\packages\microsoft.powershell.commands.management\7.3.0\runtimes\unix\lib\net7.0\Microsoft.PowerShell.Commands.Management.dll
          Project file item includes which caused reference "C:\Users\DKattan.IN\.nuget\packages\microsoft.powershell.commands.management\7.3.0\runtimes\unix\lib\net7.0\Microsoft.PowerShell.Commands.Management.dll".
            Microsoft.PowerShell.Commands.Management
1>CSC : error CS1705: Assembly 'Microsoft.PowerShell.Commands.Management' with identity 'Microsoft.PowerShell.Commands.Management, Version=7.3.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35' uses 'System.Management.Automation, Version=7.3.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35' which has a higher version than referenced assembly 'System.Management.Automation' with identity 'System.Management.Automation, Version=7.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
1>Done building project "TestPowerShell73Dependency2.csproj" -- FAILED.
```

**Comment by adityapatwardhan** [comment]
@dkattan Could you please provide the full csproj? I would like to see the target framework. 

If targeting NET6 -> use Microsoft.Powershell.SDK v 7.2.7
If targeting NET7 -> use Microsoft.Powershell.SDK v 7.3.0

v7.1.x is out of support and should not be used. Also, v7.0.x is out of support.

**Comment by dkattan** [comment]
> @dkattan Could you please provide the full csproj? I would like to see the target framework.
> 
> If targeting NET6 -> use Microsoft.Powershell.SDK v 7.2.7 If targeting NET7 -> use Microsoft.Powershell.SDK v 7.3.0
> 
> v7.1.x is out of support and should not be used. Also, v7.0.x is out of support.

[TestPowerShell73Dependency2.zip](https://github.com/PowerShell/PowerShell/files/10195602/TestPowerShell73Dependency2.zip)

csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.PowerShell.SDK" Version="7.3.0" />
  </ItemGroup>

</Project>
```

Program.cs
```csharp
// See https://aka.ms/new-console-template for more information
using System.Management.Automation.Runspaces;

Console.WriteLine("Hello, World!");
var iss = InitialSessionState.CreateDefault2();
iss.Commands.Add(new SessionStateCmdletEntry("Resolve-Path", typeof(Microsoft.PowerShell.Commands.ResolvePathCommand), null));
```
![image](https://user-images.githubusercontent.com/1424395/206723475-45f39d8c-645a-4ca5-84f3-ea107b38f8a3.png)

**Comment by dkattan** [comment]
@SeeminglyScience  I saw some work done in https://github.com/PowerShell/PowerShell/issues/18782 v7.4.0-preview.1 related to nuget packages, any chance we can get some traction on this? I just updated my test project to Microsoft.PowerShell.SDK v7.4.0-preview.1 and the problem persists.

**Comment by dkattan** [comment]
Hey @adityapatwardhan  Any update on this?

**Comment by bironeaj** [comment]
@adityapatwardhan we are having this issue too. Any update?

**Comment by adityapatwardhan** [comment]
This issue is not fixed yet. I am working on it and hopefully will be done soon. Thank you for your patience.

**Comment by bretesq** [comment]
@adityapatwardhan  We are also seeing this issue. Any luck with a timeline on this?

**Comment by dkattan** [comment]
Still an issue on 7.3.2, SDK is unusable

![image](https://user-images.githubusercontent.com/1424395/214437794-032d9e9e-27ca-4672-b9ae-baf73b3da1d6.png)

![image](https://user-images.githubusercontent.com/1424395/214437800-38960ab2-b122-40c7-ac52-485104eaee94.png)

@SeeminglyScience  @adityapatwardhan @andschwa

**Comment by dkattan** [comment]
@adityapatwardhan Can you look at my merge request and perhaps test it since you have access to the required DevOps environment? https://github.com/PowerShell/PowerShell/pull/19033

**Comment by adityapatwardhan** [comment]
This PR should fix the problem: https://github.com/PowerShell/PowerShell/pull/19302

We plan to release the next PowerShell Preview with this change.

**Comment by dkattan** [comment]
> This PR should fix the problem: #19302
> 
> We plan to release the next PowerShell Preview with this change.

Yaaass. While you’re in there, any love for SourceLink or Debug symbols? 🥺

**Comment by adityapatwardhan** [comment]
Not right now at least.  But maybe in a separate PR.

**Comment by dkattan** [comment]
> Not right now at least. But maybe in a separate PR.

Like perhaps this one you created 5 years ago?

https://github.com/PowerShell/PowerShell/issues/8184

**Comment by adityapatwardhan** [comment]
@dkattan The Microsoft.PowerShell.SDK version v7.4.0-preview.2 will be available on NuGet.org in sometime. Could you please help validate that your scenario is fixed by that?

**Comment by dkattan** [comment]
> @dkattan The Microsoft.PowerShell.SDK version v7.4.0-preview.2 will be available on NuGet.org in sometime. Could you please help validate that your scenario is fixed by that?

I'm trying but I'm not yet seeing it in nuget.

**Comment by dkattan** [comment]
> > @dkattan The Microsoft.PowerShell.SDK version v7.4.0-preview.2 will be available on NuGet.org in sometime. Could you please help validate that your scenario is fixed by that?
> 
> I'm trying but I'm not yet seeing it in nuget.

I found it, I didn't have the Include Prerelease box checked. 

Evidently it requires .NET 8 which requires Visual Studio 2022 17.6 Preview. Got all that installed.

When I attempt to install Microsoft.PowerShell.SDK or Microsoft.PowerShell.Commands.Utility I get:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1102	Unable to find package Microsoft.CodeAnalysis.CSharp with version (>= 4.6.0-2.23152.6)
  - Found 142 version(s) in nuget.org [ Nearest version: 4.6.0-1.final ]
  - Found 0 version(s) in Microsoft Visual Studio Offline Packages	TestPowerShell73Dependency2	C:\Users\DKattan\source\repos\TestPowerShell73Dependency2\TestPowerShell73Dependency2\TestPowerShell73Dependency2.csproj	1	
```
System.Management.Automation, Microsoft.PowerShell.Commands.Diagnostics and Microsoft.PowerShell.Commands.Management install fine.

Installing Microsoft.PowerShell.Commands.Management 7.4.0-preview.2 does solve the problem for TestPathCommand.

**Comment by adityapatwardhan** [comment]
The was fixed by https://github.com/PowerShell/PowerShell/pull/19302 and should be available on the latest versions of v7.2, v7.3 and v7.4.0-preview.X

Please re-open if you still see the issue.

**Comment by dkattan** [comment]
> The was fixed by #19302 and should be available on the latest versions of v7.2, v7.3 and v7.4.0-preview.X
> 
> Please re-open if you still see the issue.

Nuget package microsoft.powershell.sdk 7.3.4 is including refs to 7.3.3 causing  my project to not build

![image](https://github.com/PowerShell/PowerShell/assets/1424395/13bfb3d0-affb-4c11-a80d-b19dd2b341f0)

This appears to be fixed in 7.4.0-preview.3 but we aren't on .net 8 yet, can we get this backported?

![image](https://github.com/PowerShell/PowerShell/assets/1424395/38e419ab-359e-4bd3-a373-0701dc8ed9e7)

**Comment by dkattan** [comment]
> > The was fixed by #19302 and should be available on the latest versions of v7.2, v7.3 and v7.4.0-preview.X
> > Please re-open if you still see the issue.
> 
> Nuget package microsoft.powershell.sdk 7.3.4 is including refs to 7.3.3 causing my project to not build
> 
> ![image](https://user-images.githubusercontent.com/1424395/246097891-13bfb3d0-affb-4c11-a80d-b19dd2b341f0.png)
> 
> This appears to be fixed in 7.4.0-preview.3 but we aren't on .net 8 yet, can we get this backported?
> 
> ![image](https://user-images.githubusercontent.com/1424395/246098824-38e419ab-359e-4bd3-a373-0701dc8ed9e7.png)

Turns out the version mismatch was not actually causing a problem. 

The full error was a BadImageFormatException from ReinforcedTypings 

```
Reinforced.Typings : Assembly loading warning RT0014: Assembly Microsoft.PowerShell.Commands.Utility (from C:\Users\DKattan\.nuget\packages\microsoft.powershell.sdk\7.3.4\ref\net7.0\Microsoft.PowerShell.Commands.Utility.dll) failed to load: System.BadImageFormatException: Could not load file or assembly 'Microsoft.PowerShell.Commands.Utility, Version=7.3.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Reference assemblies cannot be loaded for execution. (0x80131058)

Reinforced.Typings : Unexpected error RT0999: Could not load file or assembly 'Microsoft.PowerShell.Commands.Utility, Version=7.3.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
```

This was not preventing the build from working, simply preventing our TypeScript from being generated post build.

I believe this would have actually been an error in 7.4 had I actually upgraded our project. 

ReinforcedTypings has some [known issues](https://github.com/search?q=repo%3Areinforced%2FReinforced.Typings+BadImageFormatException&type=issues) with reference packages but I was able to workaround it by adding Microsoft.PowerShell.Commands.Utility and System.Management.Automation nuget packages to the project containing Reinforced Typings, generating path property to them and then referencing that in the ReinforcedTypings.Config.xml so it could find the unix/win (non-reference) assembly.

No action necessary.
